streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -663,18 +666,21 @@ public boolean isProcessable(final long wallClockTime) {
      */
     @SuppressWarnings("unchecked")
     public boolean process(final long wallClockTime) {
-        if (!isProcessable(wallClockTime)) {
-            return false;
-        }
+        if (record == null) {
```

Comment:
If we have a cached record, it implies we failed to process it before -- thus we don't pull a new record from the buffer but retry to process the cached record.